### PR TITLE
Multiple assertions on the same contract

### PR DESCRIPTION
**Pull request by mikekeke**
Solves #51

### CODE REVIEW

File: src/Test/Plutip/Contract.hs [hunk 1]
```diff
@@ -113,176 +160,70 @@ type TestContractConstraints (w :: Type) (e :: Type) (a :: Type) =
...
+maybeAddValuesCheck ioRes tws =
+  bool id (valuesCheckCase :) (any isJust expected)
   where
-    mkPredicate p = \case
-      ContractExecutionError e -> p e
-      _ -> False
+    expected = twExpected <$> unTestWallets tws
 
-{- | Assert the failure reason of contract execution with a custom predicate.
+    valuesCheckCase = testCase "Values check" $ do
```
Comment (gege251) [inline src/Test/Plutip/Contract.hs:218]: I think we don't need `do` notation here, but for readability it'd be even better to replace `>>=` by something like `ExecutionResult o _ <- ioRes` (but this is more like a personal preference)

File: src/Test/Plutip/Contract.hs [hunk 2]
```diff
@@ -297,51 +238,30 @@ valueAt addr = do
     utxoValue (PublicKeyChainIndexTxOut _ v) = v
     utxoValue (ScriptChainIndexTxOut _ _ _ v) = v
 
-data TestContract (w :: Type) (e :: Type) (a :: Type) = TestContract
-  { tcContract :: TestContractConstraints w e a => TestRunner w e a
-  , tcExpected :: ExpectedOutcome w e a
-  , tcSetup :: IO (ClusterEnv, NonEmpty BpiWallet)
-  }
+-- | Test contract
+data TestContract (w :: Type) (e :: Type) (a :: Type)
```
Comment (gege251) [inline src/Test/Plutip/Contract.hs:242]: Could this be a changed to a `newtype`?

File: src/Test/Plutip/Contract.hs [hunk 3]
```diff
@@ -1,54 +1,95 @@
...
- > shouldSucceed
- >   "Two contracts after each other"
- >   (initAndAssertAdaWith 100 VLt 100 <> initAndAssertAdaWith 100 VLt 100)
- >   $ do
- >     void $
- >       withContract $
- >         \[pkh1] -> payTo pkh1 10_000_000
- >     withContractAs 1 $
- >       \[pkh1] -> payTo pkh1 10_000_000
+  wallet inside the contract, the following callback function is used together with `withContract`:
```
Comment (gege251) [inline src/Test/Plutip/Contract.hs:35]: Just checked with `cabal haddock`, but the newline doesn't seem to be rendered there. If that was your intention, could you take a look?

**Review by gege251 — COMMENTED** [review]
looking good so far

File: src/Test/Plutip/Contract.hs (hunk 3 above)
Comment (mikekeke) [reply src/Test/Plutip/Contract.hs:37]: I've revisited and updated documentation and made fixes. Haddock generated docs should be OK now.

**Review by gege251 — APPROVED** [review]

**Pull request merged**